### PR TITLE
Development

### PR DESCRIPTION
**Pull request by wejendorp**
## New
-   Made a Main/snpsearch.html template, allowing the discussed request types. (Request by search input)
-   Added data-output functions to AppRequestHandler, using new data/\* templates, including prettified output (via flag).
-   Made it possible to specify routes in the controller via **routes** module variable.
### Todo

Create specialized "controller" templates (as in rails?). Uses standard inheritance.
## Cleanup:
-   Removed unused google prettify files, and moved the essential ones to root /js dir.
-   Introducing a _Controllers_ dir, and moved all previous routed classes to this.
-   Introducing a _Models_ dir, moving model definitions to separate files.
-   Moved wikitools library into Lib dir. The Bio library requires refactoring to be moved though.
-   Merged Template.py into Utilities.py.
-   Cleaned up imports in various files.
-   Cleaned up inheritance of templates.

### CODE REVIEW

File: Controllers/databaseController.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+from xml.dom.minidom import parseString
+
+Entrez.email = 'pamad05+entrez@gmail.com'
+# Given a list of comma-seperated SNP cluster ids (example: "1805007,1805008")
+# this class queries for all the articles referenced from each of the SNP-ids and 
+# returns a list of dicts containing three values for each article:
+# - 'title' article
+# - 'abstracts' is a list of abstracts with a possible label. Ex: {label:"intro", "text":"<abstract_text>"}
+# - 'PMID' (PubMed ID) of article
+class pubmed(Utilities.AppRequestHandler):
```
Comment (paddie) [inline Controllers/databaseController.py]: gør det pænere ved at gøre følgende:

``` python
from Utilities import AppRequestHandler
class pubmed(AppRequestHandler)
```

**Comment by paddie** [comment]
Alle models specifikationerne bør være i samme fil. Især relaterede modeller (eg. snp, snp_url)

**Comment by paddie** [comment]
Hvorfor er templates/main/snpsearch.html i undermappen Main?

**Comment by paddie** [comment]
Hvis du retter de ting, og committer til dit eget repo og pusher til din profil (du behøver ikke lave et nyt pull-request, det klarer den selv), så ser det virkelig godt ud makker. Godt arbejde!

**Comment by wejendorp** [comment]
models i én fil - why?

Main/snpsearch.html er i sin egen mappe fordi det hører under mainController :) Tjek "controller-notes.pdf" i dropbox. Det er selvfølgelig bare et bud på struktur though.

Fixer Utilities importen.

**Comment by wejendorp** [comment]
Now the template acts as one would expect.

**Comment by jensraaby** [comment]
why is the pull request closed but not merged in?

**Comment by paddie** [comment]
What? I didn't close it?

Hilsen

Patrick-Ranjit D. Madsen

Sent from my mobile. Please excuse any and all typos, errors,
formatting, lack of punctuation or short responses.

On 24/03/2012, at 18.15, jensraaby
reply@reply.github.com
wrote:

**Comment by wejendorp** [comment]
Its me, I am the idiot ^^ I'll merge and close.

**Pull request merged**